std/experimental/allocator/gc_allocator.d

```diff
@@ -159,3 +159,99 @@ unittest
     // anything above a page is simply rounded up to next page
     assert(GCAllocator.instance.goodAllocSize(4096 * 4 + 1) == 4096 * 5);
 }
+
+/**
+A useful stub that only uses D builtin operations and can be used to built a
+`@safe`, `nothrow` and `pure` API with D's default garbage collector.
+At the moment it only can only be used to allocate arrays via `makeArray`.
+*/
+struct GCSafeAllocator
+{
+    enum uint alignment = platformAlignment;
+
+    void[] allocate(size_t bytes) const
+    {
+        assert(0, "stub - should not be called");
```

Comment:
This isn't a template. Static assert is useful when you want to prevent somebody from instantiating a template, but this is a normal function, therefore the static assert will always trigger.